File: Algebra_MixedToImproper.vb
Module Algebra_mixed2improper

    Sub Main()

        ''
        ' Convert fractions from Mixed To Improper
        ''
        Dim fraction As New Dictionary(Of String, Integer)
        fraction.Add("whole_number", 3)
        fraction.Add("numerator", 1)
        fraction.Add("denominator", 3)

        Console.WriteLine("    Converting from Mixed to Improper the fraction:")
        ' Print As fraction
        Console.WriteLine(String.Format("{0,55}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,53} {1}", fraction.Item("whole_number"), "-"))
        Console.WriteLine(String.Format("{0,55}", fraction.Item("denominator")))

        ' use the MixedToImproper Class
        Dim mix2imp As New MixedToImproper
        mix2imp._init_(fraction)
        fraction.Item("numerator") = mix2imp.doConvert()

        Console.WriteLine(Environment.NewLine)

        Console.WriteLine(String.Format("{0,52}", fraction.Item("numerator")))
        Console.WriteLine(String.Format("{0,52}", "Answer =      -"))
        Console.WriteLine(String.Format("{0,52}", fraction.Item("denominator")))

    End Sub

End Module

File: Algebra_MixedToImproper.cs
using System;
using System.Collections;

namespace Algebra_CS
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to our demonstration sequels");
            Console.WriteLine("Hope you enjoy (and follow) the lessons.");
            Console.WriteLine("\r\n");

            /*
            * Convert fractions from Mixed to Improper
             */
            Hashtable fraction = new Hashtable();
            fraction["whole_number"] = 3;
            fraction["numerator"] = 1;
            fraction["denominator"] = 3;

            Console.WriteLine("    Converting from Mixed to Improper the fraction:");
            // Print as fraction
            Console.WriteLine(String.Format("{0,55}", fraction["numerator"]));
            Console.WriteLine(String.Format("{0,53} {1}", fraction["whole_number"], "-"));
            Console.WriteLine(String.Format("{0,55}", fraction["denominator"]));

            // use the MixedToImproper class
            MixedToImproper mix_imp = new MixedToImproper(fraction);
            fraction["numerator"] = mix_imp.doConvert();

            Console.WriteLine(Environment.NewLine);

            Console.WriteLine(String.Format("{0,52}", fraction["numerator"]));
            Console.WriteLine(String.Format("{0,52}", "Answer =      -"));
            Console.WriteLine(String.Format("{0,52}", fraction["denominator"]));
        }
    }
}
